test/parallel/test-net-listen-shared-ports.js

```diff
@@ -42,22 +42,23 @@ if (cluster.isMaster) {
   const server1 = net.createServer(common.noop);
   const server2 = net.createServer(common.noop);
 
-  server1.on('error', function(err) {
+  server1.on('error', (err) => {
     // no errors expected
     process.send('server1:' + err.code);
   });
 
-  server2.on('error', function(err) {
+  server2.on('error', (err) => {
     // an error is expected on the second worker
     process.send('server2:' + err.code);
   });
 
   server1.listen({
     host: 'localhost',
-    port: common.PORT,
+    port: 0,
     exclusive: false
-  }, function() {
-    server2.listen({port: common.PORT + 1, exclusive: true}, function() {
+  }, () => {
+    const port2 = server1.address().port + 1;
+    server2.listen({port: port2, exclusive: true}, () => {
```

Comment:
I think there's still a problem here. There's not guarantee that `port2` is free when you call listen. See https://github.com/nodejs/node/pull/12452#discussion_r111692779 for a similar issue